flux-refineck/src/constraint_gen.rs

```diff
@@ -63,12 +63,17 @@ struct InferCtxt<'a, 'tcx> {
 #[derive(PartialEq, Eq, Clone, Copy, Hash)]
 pub struct Tag {
     pub reason: ConstrReason,
-    pub span: Span,
+    pub src_span: Span,
+    pub dst_span: Option<ESpan>,
```

Comment:
The names `src` and `dst` doesn't make too much sense to me, but I don't have anything better